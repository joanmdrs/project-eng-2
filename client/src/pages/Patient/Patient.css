.container-patient {
  display: flex;
  background-color: #d0d6db;
  min-height: 100vh;
}

.box-patient {
  width: 100%;
}

.box-header-patient {
  padding: 10px;
  padding-bottom: 5px;
  background-color: #fff;
}

.box-icon-and-title {
  display: flex;
  align-items: baseline;
  gap: 10px;
}

.box-list-functions {
  color: #848484;
  font-size: 18px;
}

.box-icon-and-title .icon {
  font-size: 25px;
}

.box-patient .form-patient {
  padding: 30px;
  margin: 10px 15px 15px 15px;
  border: 1px solid #d0d6db;
  border-radius: 4px;
  background-color: #fff;
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
}

.box-patient .input-search-patient {
  padding: 10px;
  padding-left: 0px;
  margin: 10px 15px -10px 15px;
  width: 400px;
}

.input-textarea {
  height: 180px;
  
}

.col-btn-save-cancel {
  display: flex;
  gap: 10px;
  align-items: flex-end;
  justify-content: flex-end;
}

.box-patient  .tablePatient {
  padding: 30px;
  margin: 10px 15px 15px 15px;
  border: 1px solid #d0d6db;
  border-radius: 4px;
  background-color: #fff;
  box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
  width: 1270px;
}
