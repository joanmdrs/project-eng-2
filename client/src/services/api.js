import axios from "axios";

let storedToken = localStorage.getItem('token');
let token

storedToken !== null ? token = storedToken.substring(1, storedToken.length - 1) : token = null

const api = axios.create({
  baseURL: "http://localhost:3001",
  headers: {'Authorization': 'Bearer '+ token}
});

export default api;