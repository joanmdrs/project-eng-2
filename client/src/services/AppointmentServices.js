import api from "./api"
import Swal from 'sweetalert2';


export const getValuesInput = () => {
    return {
       title: document.getElementById("title").value,
       patient_cpf: document.getElementById("patient-cpf").value,
       patient_name: document.getElementById("patient-name").value,
       doctor_crm: document.getElementById("doctor-crm").value,
       doctor_name: document.getElementById("doctor-name").value,
       data_appointment: document.getElementById("data-appointment").value,
       hour_appointment: document.getElementById("hour-appointment").value,
       description: document.getElementById("description").value
   };
 }
 
 export const getActionForm = () => {
    return document.getElementById("form-appointment").dataset.action;
 }
 
 export const addAppointment = (data) => {
 
    const { 
      title, 
      patient_cpf, 
      patient_name, 
      doctor_crm, 
      doctor_name, 
      data_appointment, 
      hour_appointment, 
      description} = data;

    return api.post("/appointments", {
      title: title,
      patient_cpf: patient_cpf,
      patient_name: patient_name,
      doctor_crm: doctor_crm,
      doctor_name: doctor_name,
      data_appointment: data_appointment,
      hour_appointment: hour_appointment,
      description: description
    });
}

export const getAppointments = async () => {
    const res = await api.get("/appointments");
    const data = res.data;    
    return JSON.stringify(data);
 }


 export const setFields = (data) => {

    const {
      id, 
      title, 
      patient_cpf, 
      patient_name, 
      doctor_crm, 
      doctor_name, 
      data_appointment, 
      hour_appointment, 
      description} = data;
 
    document.getElementById("form-appointment").dataset.action = id;
    document.getElementById("title").value = title;
    document.getElementById("patient-cpf").value = patient_cpf;
    document.getElementById("patient-name").value = patient_name;
    document.getElementById("doctor-crm").value = doctor_crm;
    document.getElementById("doctor-name").value = doctor_name;
    document.getElementById("data-appointment").value = data_appointment;
    document.getElementById("hour-appointment").value = hour_appointment;
    document.getElementById("description").value = description;
 }
 
 export const updateAppointment = (id, data) => {
 
    const {
      title, 
      patient_cpf, 
      patient_name, 
      doctor_crm, 
      doctor_name, 
      data_appointment, 
      hour_appointment, 
      description} = data;
 
    return api.put(`/appointments/${id}`, {
      id: id,
      title: title,
      patient_cpf: patient_cpf,
      patient_name: patient_name,
      doctor_crm: doctor_crm,
      doctor_name: doctor_name,
      data_appointment: data_appointment,
      hour_appointment: hour_appointment,
      description: description
    });    
}

export const deleteAppointment = (id) => {

  Swal.fire({
    title: 'Are you sure?',
    text: "You won't be able to revert this!",
    icon: 'warning',
    showCancelButton: true,
    confirmButtonColor: '#d33',
    cancelButtonColor: '#0C6170',
    confirmButtonText: 'Yes, delete it!'
  }).then((result) => {
    if (result.isConfirmed) {
      api.delete(`/appointments/${id}`);
      document.location.reload();
    }
  });
}

export const messageSucess = (message) => {
  Swal.fire({
    title: 'Success',
    text: message,
    icon: 'success',
    showCancelButton: false,
    confirmButtonColor: '#0C6170',
    confirmButtonText: 'Ok',
  }).then((result) => {
    if (result.isConfirmed) {
      document.location.reload();
    }
  });
}

export const messageFailure = (message) => {
  Swal.fire({
    title: "Failure",
    text: message,
    icon: "error",
    showCancelButton: false,
    confirmButtonColor: "#0C6170",
    confirmButtonText: "Ok",
  }).then((result) => {
    if (result.isConfirmed) {
      document.location.reload();
    }
  });
}
 
export const messagePrepareToUpdate = (title, text) => {
  Swal.fire({
    title: title,
    text: text,
    icon: 'info',
    showCancelButton: false,
    confirmButtonColor: '#0C6170',
    confirmButtonText: 'Ok',
  });
}

