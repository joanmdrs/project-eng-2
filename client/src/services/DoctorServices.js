import api from "./api";
import Swal from "sweetalert2";

export const addDoctor = (data) => {
    const {name, crm, phone, email, username, password} = data;

    api.post("/doctors", {
      name: name,
      crm: crm,
      phone: phone,
      email: email,
      username: username,
      password: password
    });
}

export const getDoctorByCrm = async (crmProvided) => {
    const res = await api.get(`/doctors/${crmProvided}`);
    const data = res.data;
    return JSON.stringify(data);
}

export const updateDoctor = (id, data) => {
    const {name, crm, phone, email, username, password} = data;

    api.put(`/doctors/${id}`, {
      id: id,
      name: name,
      crm: crm,
      phone: phone,
      email: email,
      username: username,
      password: password
    });
}

export const deleteDoctor = (id) => {
    api.delete(`/doctors/${id}`);
}

export const getDoctors = async () => {
    const res = await api.get("/doctors");
    const data = res.data;
    return JSON.stringify(data);
}

export const getValuesInput = () => {
    const data = {
       name: document.getElementById("input-name").value,
       crm: document.getElementById("input-crm").value,
       phone: document.getElementById("input-phone").value,
       email: document.getElementById("input-email").value,
       username: document.getElementById("input-username").value,
       password: document.getElementById("input-password").value
   };
   return data;
}

export const getActionForm = () => {
    const action = document.getElementById("form-doctor").dataset.action;
    return action;
}

export const setFields = (data) => {
    const {id, name, crm, phone, email, username, password} = data;
    document.getElementById("form-doctor").dataset.action = id;
    document.getElementById("input-name").value = name;
    document.getElementById("input-crm").value = crm;
    document.getElementById("input-phone").value = phone;
    document.getElementById("input-email").value = email;
    document.getElementById("input-username").value = username;
    document.getElementById("input-password").value = password;
}

export const messagePrepareToUpdate = () => {
    Swal.fire({
       title: 'Edit Doctor',
       text: "Now you will edit this Doctor's informations, be careful.",
       icon: 'info',
       showCancelButton: false,
       confirmButtonColor: '#0C6170',
       confirmButtonText: 'Ok',
    });
}

export const messageConfirm = (message) => {
    Swal.fire({
       title: 'Success',
       text: message,
       icon: 'success',
       showCancelButton: false,
       confirmButtonColor: '#0C6170',
       confirmButtonText: 'Ok',
    }).then((result) => {
       if (result.isConfirmed) {
          document.location.reload();
       }
    });
}

export const messageFailure = (message) => {
    Swal.fire({
       title: 'Error',
       text: message,
       icon: 'error',
       showCancelButton: false,
       confirmButtonColor: '#0C6170',
       confirmButtonText: 'Ok',
    }).then((result) => {
       if (result.isConfirmed) {
          document.location.reload();
       }
    });
}
