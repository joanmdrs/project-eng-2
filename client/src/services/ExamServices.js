import api from "./api"

export const addExam = (exam) => {

    const {title, patient_cpf, doctor_crm, description} = exam;



    api.post("/exam/register", {
       title: title,
       patient_cpf: patient_cpf,
       doctor_crm: doctor_crm,
       description: description
    });
  
}
