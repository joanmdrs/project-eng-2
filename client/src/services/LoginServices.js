import api from "./api";


export const signIn = async () => {
    const username = document.getElementById("username").value;
    const password = document.getElementById("password").value;
    const response = await api.post("/auth/login", {
        username: username,
        password: password
    });

    localStorage.setItem('token', JSON.stringify(response.data));
    console.log(response);
}

export const isAuthenticate = () => {
    const token = localStorage.getItem('token');
}