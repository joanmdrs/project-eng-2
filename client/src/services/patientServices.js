import api from "./api";
import Swal from "sweetalert2";
export const createPatient = (props) => {
  let response = api.post("/patients", {
    name: props.name,
    cpf: props.cpf,
    phone: props.phone,
    email: props.email,
    username: props.username,
    password: props.password,
  });
  return response;
}

export const messageSucess = (message) => {
  Swal.fire({
    title: "Success",
    text: message,
    icon: "success",
    showCancelButton: false,
    confirmButtonColor: "#0C6170",
    confirmButtonText: "Ok",
  }).then((result) => {
    if (result.isConfirmed) {
      document.location.reload();
    }
  });
};

export const messageFailure = (message) => {
  Swal.fire({
    title: "Failure",
    text: message,
    icon: "error",
    showCancelButton: false,
    confirmButtonColor: "#0C6170",
    confirmButtonText: "Ok",
  }).then((result) => {
    if (result.isConfirmed) {
      document.location.reload();
    }
  });
};

export const listPatients = async () => {
  const res = await api.get("/patients");
  const data = res.data;    
  return JSON.stringify(data);
}

export const filterPatient = async (cpf) => {
  const res = await api.get(`/patients/${cpf}`);
  const data = res.data;    
  return JSON.stringify(data);
}

export const editPatient = (props) => {
  return api.put(`/patients/${props.id}`, {
    id: props.id,
    name: props.name,
    cpf: props.cpf,
    phone: props.phone,
    email: props.email,
    username: props.username,
    password: props.password,
  })
}

export const removePatient = (id) => {
  api.delete(`/patients/${id}`);
}