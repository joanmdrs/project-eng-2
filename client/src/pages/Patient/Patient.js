import './Patient.css';
import React, { useState, useRef, useEffect } from 'react';
import Nav from '../../components/Nav/Nav';
import 'bootstrap/dist/css/bootstrap.min.css';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHospitalUser, faPenToSquare , faTrash } from '@fortawesome/free-solid-svg-icons';
import { Form, FormGroup, Label, Input, Row, Col, Button, Table, InputGroup } from 'reactstrap';
import { createPatient, editPatient, removePatient, listPatients, filterPatient, messageSucess, messageFailure } from '../../services/patientServices.js'
import TableCard from "../../components/Table/Table.js";
import Swal from 'sweetalert2';
import Listpatient from '../../components/ListPatient/ListPatient';
import ListPatient from '../../components/ListPatient/ListPatient';


export default function Patient() {

  const [values, setValues] = useState();
  const [listValues, setListValues] = useState([]);
  const [searchValue, setSearchValue] = useState([]);




  const idRef = useRef(null);
  const nameRef = useRef(null);
  const cpfRef = useRef(null);
  const phoneRef = useRef(null);
  const emailRef = useRef(null);
  const usernameRef = useRef(null);
  const passwordRef = useRef(null);

  const handleChangeValues = (value) => {
    setValues((prevValue) => ({
      ...prevValue,
      [value.target.name]: value.target.value,
    }));
  };

  const handleSaveButton = () => {
    createPatient(values).then((patient)=>{
      messageSucess(`New Patient was added.`)
    }).catch((error)=>{
      console.log("Caiu aqui", error)
      messageFailure("Something went wrong.");
    });
    
    
  }

  const handleClickButton = () => {
    if (document.getElementById("form-patient").dataset.action === "new") {
      handleSaveButton();
    } else {
      handleUpdateButton();
    }
  };

  const handleUpdateButton = async () => {
    const id = document.getElementById("id").value;
    const name = document.getElementById("name").value;
    const cpf = document.getElementById("cpf").value;
    const phone = document.getElementById("phone").value;
    const email = document.getElementById("email").value;
    const username = document.getElementById("username").value;
    const password = document.getElementById("password").value;
    const patientUpdated = {
      id: Number(id),
      name: name,
      cpf: cpf,
      phone: phone,
      email: email,
      username: username,
      password: password,
    };
    editPatient(patientUpdated).then((response)=>{
      messageSucess(`${response.data.name} was updated.`)
    }).catch((error)=>{
      console.log(error)
       Swal.fire({
         title: "Edit Patient",
         text: "Something was wrong.",
         icon: "info",
         showCancelButton: false,
         confirmButtonColor: "#0C6170",
         confirmButtonText: "Ok",
       });
    })

    
  };

  const handleCancelButton = () => {
    idRef.current.value = '';
    nameRef.current.value = '';
    cpfRef.current.value = '';
    phoneRef.current.value = '';
    emailRef.current.value = '';
    usernameRef.current.value = '';
    passwordRef.current.value = '';
  };

  const handleFilterPatientByCpf = async ()  => {
    const cpfReceived = document.getElementById("searchCpf").value;
    const data = await filterPatient(cpfReceived);
    const patient = JSON.parse(data);
    setListValues([patient]);

  }


  useEffect(() => {
    const fetchData = async () => {
      const data = await listPatients();
      const patients = JSON.parse(data);
      setListValues(patients);
    };

    fetchData();
  }, []);

  console.log(listValues)

  return (
    <div className="container-patient">
      <Nav />
      <div className="box-patient">
        <div className="box-header-patient">
          <div className="box-icon-and-title">
            <FontAwesomeIcon className="icon" icon={faHospitalUser} />
            <h3> Patients </h3>
          </div>
          <div className="box-list-functions">
            <p>
              Patient registration: Include, Search, Change, Delete and List
            </p>
          </div>
        </div>
        <Form className="form-patient" id="form-patient" data-action="new">
          <Row>
            <Col md={8}>
              <FormGroup className='idForm'>
                <Label for="id">Id</Label>
                <Input id="id" type="text" placeholder="id" ref={idRef} onChange={handleChangeValues}></Input>
              </FormGroup>

              <FormGroup>
                <Label for="name">Name</Label>
                <Input id="name" name="name" type="text" placeholder="name" ref={nameRef} onChange={handleChangeValues}></Input>
              </FormGroup>
            </Col>
          </Row>
          <Row>
            <Col md={4}>
              <FormGroup>
                <Label for="cpf">CPF</Label>
                <Input id="cpf" name="cpf" type="text" placeholder="cpf" ref={cpfRef} onChange={handleChangeValues}></Input>
              </FormGroup>
            </Col>
            <Col md={4}>
              <FormGroup>
                <Label for="phone">Phone</Label>
                <Input id="phone" name="phone" type="text" placeholder="phone" ref={phoneRef} onChange={handleChangeValues}></Input>
              </FormGroup>
            </Col>
          </Row>
          <Row>
            <Col md={8}>
              <FormGroup>
                <Label for="email">Email</Label>
                <Input id="email" name="email" type="email" placeholder="email" ref={emailRef} onChange={handleChangeValues}></Input>
              </FormGroup>
            </Col>
          </Row>
          <Row>
            <Col md={4}>
              <FormGroup>
                <Label for="username">Userame</Label>
                <Input id="username" name="username" type="text" placeholder="username" ref={usernameRef} onChange={handleChangeValues}></Input>
              </FormGroup>
            </Col>
            <Col md={4}>
              <FormGroup>
                <Label for="password">Password</Label>
                <Input id="password" name="password" type="password" placeholder="password" ref={passwordRef} onChange={handleChangeValues}></Input>
              </FormGroup>
            </Col>
          </Row>
          <Row>
            <Col className="col-btn-save-cancel" md={12}>
              <Button 
              type="button" 
              onClick={() => {
                handleClickButton();
              }} 
              className="btn-save btn-success">
                  Save
              </Button>
              <Button 
              type="submit" 
              className="btn-cancel btn-danger"
              onClick={handleCancelButton}>
                Cancel
              </Button>
            </Col>
          </Row>
        </Form>

        <div className="input-search-patient">
          <Label for="searchCpf">Filter By CPF</Label>
          <InputGroup>
            <Input id="searchCpf" type="text" placeholder="cpf"></Input> 
            <Button 
            color="success" 
            onClick={() => {
              handleFilterPatientByCpf();
            }}>
              Search
            </Button>
          </InputGroup>
        </div>
        <ListPatient
          patients={listValues}

        >

        </ListPatient>
        

      </div>
    </div>
  )
}

export const handleDeleteButton = async (id) => {
  Swal.fire({
    title: 'Are you sure?',
    text: "You won't be able to revert this!",
    icon: 'warning',
    showCancelButton: true,
    confirmButtonColor: '#d33',
    cancelButtonColor: '#0C6170',
    confirmButtonText: 'Yes, delete it!'
  }).then((result) => {
    if (result.isConfirmed) {
      removePatient(id);
      document.location.reload();
    }
  })
};
