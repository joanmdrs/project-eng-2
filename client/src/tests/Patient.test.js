import { findPatientByCpf, getValuesFormPatient } from '../services/patientServices';
import renderer from 'react-test-renderer';
import Patient from "../pages/Patient/Patient"

let container = null;
let id = null;
let name = null;
let cpf = null;
let phone = null;
let email = null;
let username = null;
let password = null;

beforeEach(() => {
  container = document.createElement("div");

  id = document.createElement("input");
  id.setAttribute("id", "form-patient");
  id.setAttribute("data-action", "new");

  name = document.createElement("input");
  name.setAttribute("id", "name");
  name.setAttribute("value","Joan");

  cpf = document.createElement("input");
  cpf.setAttribute("id", "cpf");
  cpf.setAttribute("value","11058118404");

  phone = document.createElement("input");
  phone.setAttribute("id", "phone");
  phone.setAttribute("value","84996269651");

  email = document.createElement("input");
  email.setAttribute("id", "email");
  email.setAttribute("value","joanmdrs@gmail.com");

  username = document.createElement("input");
  username.setAttribute("id", "username");
  username.setAttribute("value", "joanmdrs");

  password = document.createElement("input");
  password.setAttribute("id", "password");
  password.setAttribute("value", "joan123");

  container.appendChild(id);
  container.appendChild(name);
  container.appendChild(cpf);
  container.appendChild(phone);
  container.appendChild(email);
  container.appendChild(username);
  container.appendChild(password);
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});


//Teste getValuesInput
it("Retornando a Descrição do Paciente", () => {

  const got = getValuesFormPatient();
  const expected = {
    id: "new",
    name: "Joan",
    cpf: "11058118404",
    phone: "84996269651",
    email: "joanmdrs@gmail.com",
    username: "joanmdrs",
    password: "joan123",
  };

  expect(expected).toEqual(got);
});

test("resultados devem ser idênticos", async () => {
  const data = await findPatientByCpf('11058118404')
  const patients = JSON.parse(data)
  expect(patients.name).toBe('Allan')
}) 


it('testando a renderização a page Patient', () => {
  const component = renderer.create(Patient);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot(); 
})