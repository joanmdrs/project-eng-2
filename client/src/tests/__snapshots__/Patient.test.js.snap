// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando a renderização a page Patient 1`] = `null`;
